_csharp-api/inputsystem/input.md

```diff
@@ -2,8 +2,60 @@
 
 This static type provides you with input functionality, the base input system has no events to bind to and supports polling only. It's largely self explanatory with a few possible gotchas.
 
-## Key Polling
+## Key Reading
 
-`GetKeyDown(KeyCode)` | `GetKeyPressed(KeyCode)` | `GetKeyReleased(KeyCode)` - These are likely to be the most commonly used methods, `GetKeyDown` will return true on any number of frames so long as a key is down, this can cause undesired behaviour. If you only want to read an input once, use `GetKeyPressed` or `GetKeyReleased`.
+There's a number of usefull methods that detect keys being pressed down in different ways, all seeming to return a 'bool'. Most common are `isKeyDown`, `isKeyPressed` and `isKeyReleased`.
 
-You can also use the extension methods `IsDown`/`IsPressed`/`IsReleased` like so `KeyCode.A.IsPressed()`
\ No newline at end of file
+You may also use the extension methods `isDown`, `isPressed` and `isReleased` as *extension* alternatives of the above
+
+Below, you may find an in-depth documentation of each.
+
+ ### Input.IsKeyDown(InputKey.[key])
```

Comment:
Missing return type

Reply:
What do you mean?

Reply:
Hm? Already mentioned it returns a bool

Reply:
Use full annotation for functions.
For example.
`public virtual bool Equals (object obj);`
`Input.` can be ommited, because this file only contains methods and properties of Input class.
And clearify `InputKey.[key]`. What its type and parameter name.

Reply:
Annotations? Can't find any mention of such a thing in the docs and I'm not a web dev wizard :c. Also, afaik, even if you imported InputSystem, not using `Input.` will get you into an error.

Will specify the type of inputkey (even though it's kind of obvious?)

Reply:
Look at https://docs.bannerlordmodding.com/_csharp-api/campaignsystem/campaigngamestarter.html
Methods.
There are methods modifiers such as `public` and return type.

How you import InputSystem?

Type of `InputKey.[key]` not obvious at all. Is that build Windows Key type? Or TW wrappers on it? If second - refference to docs of that class would be usefull.

Reply:
Man that's the first time I hear someone reefer to it as "Annotation"

There's no need to declare the security level in this case, like not at all. The return type is always bool.

using Taleworlds.InputSystem;

Either that or string.

Reply:
Excuse me for asking, but are you by any chance new to Game development in C#?

Reply:
1) Sorry. Annotation often uses in that case in dynamic-typed languages.
2) Need. If it have `protected` it will impossible to use it without inheritance.
3) Try `using Taleworld.InputSystem.Input;`
4) So typed it, please.

Reply:
1. I get it, but I never saw someone call them "Annotations", tbh never actually know how it's called.
2. It doesn't tho... it's built for this very purpose, to be used as a condition easily
3. You can do that but why assume a beginner would understand what's wrong?
4. aight

man after this fix just publish it, I will add more changes as I test stuff more but the current version that is on the website is complete gibberish that is not included in the API, and while we argue here someone might need to add input functionality but can't because the don't know how...

Reply:
3) Rider support autoimport. And if you write on C# - you should now basics of. There is no other way.
I'm not able to merge it, anyway, because I am not maintainer of this repo. ._.